APMrover2/mode.h

```diff
@@ -280,6 +280,51 @@ class ModeGuided : public Mode
     float _desired_yaw_rate_cds;// target turn rate centi-degrees per second
 };
 
+class ModeBreadcrumb : public ModeGuided
+{
+public:
+
+    uint32_t mode_number() const override { return BREADCRUMB; }
+    const char *name4() const override { return "BREADCRUMB"; }
+
+    // methods that affect movement of the vehicle in this mode
+    void update() override;
+
+    // attributes of the mode
+    bool is_autopilot_mode() const override { return true; }
+
+    void mavlink_packet_received(const mavlink_message_t &msg);
+
+
+protected:
+
+    bool _enter() override;
+
+    void _exit()  override;
+
+private:
+
+    uint8_t target_srcid = 2; //#TODO maybe make a user parameter?
+
+    Location target_loc;
+    Vector3f target_vel;
+
+    Location *_current_crumb; //waypoint crumb we are going to
+    Location *_crumbs;    //pointer to circular queue that stores points that target vehicle crossed
+
+    const uint16_t crumbs_max = 500;  // maximum size of the queue #TODO maybe make a user parameter?
+
+    uint16_t _crumbs_tail = 0; //end of the queue
```

Comment:
We tend to leave off initialisers unless they're non-zero.  Anything in a mode object is going to be in the bss which is zeroed on start.

Leaving these in will cost flash for no benefit :-)